src/Catalog.API/Apis/CatalogApi.cs

```diff
@@ -126,38 +125,21 @@ public static IEndpointRouteBuilder MapCatalogApi(this IEndpointRouteBuilder app
 
         if (!services.CatalogAI.IsEnabled)
         {
-           return await GetItemsByName(paginationRequest, services, text);
+            return await GetItemsByName(paginationRequest, services, text);
         }
 
-        // Create an embedding for the input search
-        var vector = await services.CatalogAI.GetEmbeddingAsync(text);
-
         // Get the total number of items
         var totalItems = await services.Context.CatalogItems
             .LongCountAsync();
 
         // Get the next page of items, ordered by most similar (smallest distance) to the input search
-        List<CatalogItem> itemsOnPage;
-        if (services.Logger.IsEnabled(LogLevel.Debug))
-        {
-            var itemsWithDistance = await services.Context.CatalogItems
-                .Select(c => new { Item = c, Distance = c.Embedding.CosineDistance(vector) })
-                .OrderBy(c => c.Distance)
-                .Skip(pageSize * pageIndex)
-                .Take(pageSize)
-                .ToListAsync();
+        List<CatalogItem> itemsOnPage = [];
+        var itemsWithDistance = services.CatalogAI.SearchMemoryAsync(text, pageSize);
 
-            services.Logger.LogDebug("Results from {text}: {results}", text, string.Join(", ", itemsWithDistance.Select(i => $"{i.Item.Name} => {i.Distance}")));
-
-            itemsOnPage = itemsWithDistance.Select(i => i.Item).ToList();
-        }
-        else
+        await foreach (var item in itemsWithDistance)
```

Comment:
Noting that in addition to losing pagination and the distance, this performs an additional database roundtrip for each result (so for a page size of 5, this does one query via SearchMemoryAsync, and then 5 FindAsync invocations totaling 6)... That's not great in terms of performance.